Ambit/Source/Ambit/Mode/GltfExport.spec.cpp

```diff
@@ -36,51 +37,54 @@ void GltfExportSpec::Define()
         {
             World = FAutomationEditorCommonUtils::CreateNewMap();
 
-            Exporter = NewObject<UMockableGltfExport>();
+            ExternalExporter = NewObject<UGltfExporterExternalMock>();
+
+            Exporter = NewObject<UGltfExportImpl>();
+            Exporter->SetMockObjects(ExternalExporter);
+        });
+
+        Describe("Test GLTFExporter plugin not installed", [this]()
+        {
+            BeforeEach([this]()
+            {
+                ExternalExporter->SetOutputs(false, false, false);
+            });
+
+            It("Should return error if Exporter doesn't exist", [this]()
+            {
+                const GltfExportReturnCode ReturnCode = Exporter->Export(World, Filename);
+
+                TestEqual("Export should fail", ReturnCode, ExporterNotFound);
+            });
         });
 
         Describe("Test GLTFExporter plugin failure", [this]()
         {
             BeforeEach([this]()
             {
-                auto MockFunction = [](UGLTFLevelExporter* LevelExporter, UWorld* World, FBufferArchive& Buffer) -> bool
-                {
-                    return false;
-                };
-                Exporter->SetExportBinary(MockFunction);
+                ExternalExporter->SetOutputs(true, false, false);
             });
 
             It("Should return error if ExportBinary() fails", [this]()
             {
-                const UGltfExport::GltfExportReturnCode ReturnCode = Exporter->Export(World, Filename);
+                const GltfExportReturnCode ReturnCode = Exporter->Export(World, Filename);
 
-                TestEqual("Export should fail", ReturnCode, UGltfExport::Failed);
+                TestEqual("Export should fail", ReturnCode, Failed);
             });
         });
 
         Describe("Test write to file", [this]()
         {
             BeforeEach([this]()
             {
-                auto MockBinaryFunction = [](UGLTFLevelExporter* LevelExporter, UWorld* World,
-                                             FBufferArchive& Buffer) -> bool
-                {
-                    return true;
-                };
-                Exporter->SetExportBinary(MockBinaryFunction);
-
-                auto MockWriteFunction = [](FBufferArchive& Buffer, const FString& Filename) -> bool
-                {
-                    return false;
-                };
-                Exporter->SetWriteToFile(MockWriteFunction);
+                ExternalExporter->SetOutputs(true, true, false);
             });
 
             It("Should fail if file name is empty", [this]()
             {
-                const UGltfExport::GltfExportReturnCode ReturnCode = Exporter->Export(World, Filename);
+                const GltfExportReturnCode ReturnCode = Exporter->Export(World, Filename);
```

Comment:
You're making an assumption here that `Filename`—which is a persistent variable of this test spec—won't have been set by some other test. That's not a safe assumption to make. I recommend setting `Filename` to an empty string or null value in the `BeforeEach()` that runs before this test.